utils/src/main/java/uk/gov/pay/commons/utils/xray/XRaySessionProfiler.java

```diff
@@ -69,34 +70,52 @@ public Object profileExecutionOfQuery(DatabaseQuery databaseQuery, Record record
-            DatabaseMetaData metadata = connection.getMetaData();
-            databaseMetadata.put("url", metadata.getURL());
-            databaseMetadata.put("user", metadata.getUserName());
-            databaseMetadata.put("driver_version", metadata.getDriverVersion());
-            databaseMetadata.put("database_type", metadata.getDatabaseProductName());
-            databaseMetadata.put("database_version", metadata.getDatabaseProductVersion());
-            databaseMetadata.put("preparation", databaseQuery.isCallQuery() ? "call" : "statement");
-            databaseMetadata.put("sanitized_query", StringUtils.isEmpty(databaseQuery.getSQLString()) ? "" : databaseQuery.getSQLString());
-            return databaseMetadata;
-        } catch (SQLException exception) {
-            logger.warn("Error getting database connection details.");
+            DatabaseLogin datasourceLogin = (DatabaseLogin) abstractSession.getDatasourceLogin();
+            return Optional.ofNullable(datasourceLogin);
+        } catch (ClassCastException e) {
+            logger.warn("Cannot cast to DatabaseLogin [{}]", e.getMessage());
+            return Optional.empty();
         }
-        return Collections.emptyMap();
     }
 
-    private String getDatabaseHostName(Connection connection) {
-        try {
-            DatabaseMetaData metadata = connection.getMetaData();
-            String hostname = new URI((new URI(metadata.getURL())).getSchemeSpecificPart()).getHost();
-            return connection.getCatalog() + "@" + hostname;
-        } catch (URISyntaxException exception) {
-            logger.warn("Error parsing database host name.");
-        } catch (SQLException exception) {
-            logger.warn("Error getting database connection details.");
+    private Map<String, Object> getQueryMetadataWithLogin(DatabaseLogin databaseLogin, DatabaseQuery databaseQuery) {
+        Map<String, Object> databaseMetadata = getQueryMetadata(databaseQuery);
+        databaseMetadata.putAll(getLoginDetails(databaseLogin));
+        return databaseMetadata;
+    }
+
+    private Map<String, Object> getQueryMetadata(DatabaseQuery databaseQuery) {
+        Map<String, Object> databaseMetadata = new HashMap<>();
+        databaseMetadata.put("preparation", databaseQuery.isCallQuery() ? "call" : "statement");
+        databaseMetadata.put("sanitized_query", StringUtils.isEmpty(databaseQuery.getSQLString()) ? "" : databaseQuery.getSQLString());
+        return databaseMetadata;
+    }
+
+    private Map<String, Object> getLoginDetails(DatabaseLogin databaseLogin) {
+        Map<String, Object> databaseMetadata = new HashMap<>();
+        if (null != databaseLogin) {
+            databaseMetadata.put("url", databaseLogin.getURL());
+            databaseMetadata.put("user", databaseLogin.getUserName());
+        }
+        return databaseMetadata;
+    }
+
+    private String getDatabaseHostName(DatabaseLogin databaseLogin) {
+        if (null != databaseLogin) {
```

Comment:
Don't think the `null` check is necessary since if its `null` the `Optional` returned by `getDatabaseLogin` will be empty and `.map(this::getDatabasehostName)` will not be called.